### ISSUE TITLE
Custom concrete mixer recipes not working.

### ISSUE THREAD

**Issue by Electrofried** [ldtteam/minecolonies-features#448]
### Prerequisites 

- [x ] I am running the latest **alpha** version of MineColonies and Structurize for my Minecraft version.
- [x ] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure my issue is not covered there.
- [x ] I made sure that **this issue is not a duplicate of any existing issue.**

### Context 

- Minecraft Version: 1.16.5
- MineColonies Version: 0.14.182-ALPHA
- Structurize Version: 0.13.174-ALPHA
- Related Mods and their Versions: n/a

## Expected behavior

Custom crafter recipes loaded by datapack should function as they do for other crafters. The concrete mixer already has crafter functions for making concrete powder that indicate the functionality is there.

## Actual behavior

The concrete mixer refuses to perform the craft, possibly due to attempting to do it as in world crafting.  The task appears but stalls at the end and never completes. 

## Steps to reproduce the issue

1. stick a custom json recipe in a datapack such as the one attached below.
2. request said item and provide supplies and a concrete mixer
3. The worker will fail to execute the craft

### Logs

- latest.log: 
- crashlog: 

### Notes
If this is a bug related to in world crafting as Mekle indicated on discord, perhaps separating
 the 'in world' aspect of the recipe to another "type" other than recipe, split the place in world function to "water_interaction" or some other recipe type allowing the mixer to distinguish between the different types of crafting. 

<details><summary>Example JSON</summary>

```JSON
{
    "type": "recipe",
    "crafter": "concretemixer",
    "inputs": [
      {
        "item": "minecraft:sand",
        "count": 8
      },
      {
        "item": "minecraft:gravel",
        "count": 4
      },
      {
        "item": "minecraft:clay",
        "count": 1
      }
    ],
    "result": "immersiveengineering:concrete",
    "count": 16,
    "intermediate": "minecraft:air",
    "min-building-level": 1,
    "show-tooltip": true
  }

```

</details>

---
## Viewers

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by Raycoms** [comment]
Well, for the time being, the concrete mixer can do two things. A) Concrete powder b) Concrete from poweder.
That was never different. There is special code that stops him from doing anything besides that.

So this is more of a feature request allowing them to make concrete types that having nothing to do with how vanilla concrete works. To be exact, you could teach this recipe to the mechanic and the mechanic would be able to handle it just fine.

**Comment by uecasm** [comment]
Yeah, there currently isn't any general "apply item to water in-world to get other item" sort of crafting supported, it's special-cased explicitly for concrete and won't work for any other type of item.  What recipes were you trying to add?

**Comment by Electrofried** [comment]
I was trying to add just simple crafter recipes just like the concrete powder ones. I attached an example json. 
The problem is the mixer seems to have some issues with basic recipes as it has this special functionality in built. This is why I suggested that those recipes that do require the special functions built in to the mixer hut be sort of quarantined off to a different recipe type in the json like  "type": "concrete". Then when the mixer gets a standard ""type": "recipe", they will just be able to craft it as every other hut can.
